src/components/DatePicker/CalenderDatePicker.tsx

```diff
@@ -0,0 +1,248 @@
+/*!
+ * © 2023 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import 'react-modern-calendar-datepicker/lib/DatePicker.css'
+
+import { add, format, intervalToDuration } from 'date-fns'
+import React, { useCallback, useMemo, useState } from 'react'
+import DatePicker, {
+  Day,
+  DayValue,
+  RenderInputProps,
+} from 'react-modern-calendar-datepicker'
+import styled from 'styled-components'
+
+import { Category, Dialog, MessageContainer } from '../Dialog'
+export const calenderDatePickerDefaultDateFormat = 'd MMM, EEEE'
+
+const getDay = (date: Date) => ({
+  year: date.getFullYear(),
+  month: date.getMonth() + 1,
+  day: date.getDate(),
+})
+
+const potentialDueDate = (
+  stepDueDate: Date,
+  dueDate: Day,
+  submissionDueDate: Date
+) => {
+  const duration = intervalToDuration({
+    start: stepDueDate,
+    end: new Date(
+      `${dueDate.year}-${
+        dueDate.month < 10 ? `0${dueDate.month}` : dueDate.month
+      }-${dueDate.day < 10 ? `0${dueDate.day}` : dueDate.day}`
+    ),
+  })
+  return add(submissionDueDate, duration)
+}
+
+export type DatePickerConfigs = {
+  color?: string
+  position?: 'auto' | 'top' | 'bottom'
+  calendarClassName?: string
+  selectedDayClassName?: string
+}
+
+export type CalenderDialogConfigs = {
+  header?: string
+  dueDateMessage?: string
+}
+
+export type CalenderDatePicker = {
+  currentDueDate: Date
+  originalDueDate: Date
+  handleDateChange: (day: DayValue) => void
+  datepickerConfigs?: DatePickerConfigs
+  dialogConfigs?: CalenderDialogConfigs
+  Button: React.FC<RenderInputProps>
+  primaryButtonTitle?: string
+  secondaryButtonTitle?: string
+}
+
+export const CalenderDatePicker: React.FC<CalenderDatePicker> = ({
+  Button,
+  handleDateChange,
+  originalDueDate,
+  currentDueDate,
+  datepickerConfigs,
+  dialogConfigs,
+  primaryButtonTitle = 'Reschedule',
+  secondaryButtonTitle = 'Cancel',
+}) => {
+  const datePickerConfigs: DatePickerConfigs = Object.assign(
```

Comment:
It was for type-safety I thought it would trigger a Runtime error if I used the variable without checking if it's an undefined object.
I've used the spread operator and it works fine.